src/pybind/mgr/telemetry/module.py

```diff
@@ -187,81 +200,95 @@ def config_notify(self):
 
-        self.report_id = self.get_store('report_id', None)
-        if self.report_id is None:
+        report_id = self.get_store('report_id', None)
+        if report_id is None:
             self.report_id = str(uuid.uuid4())
             self.set_store('report_id', self.report_id)
+        else:
+            self.report_id = report_id
 
-        self.salt = self.get_store('salt', None)
-        if not self.salt:
+        salt = self.get_store('salt', None)
+        if salt is None:
             self.salt = str(uuid.uuid4())
             self.set_store('salt', self.salt)
+        else:
+            self.salt = salt
 
-    def gather_osd_metadata(self, osd_map):
+    def gather_osd_metadata(self,
+                            osd_map: Dict[str, List[Dict[str, int]]]) -> Dict[str, Dict[str, int]]:
         keys = ["osd_objectstore", "rotational"]
         keys += self.metadata_keys
 
-        metadata = dict()
+        metadata: Dict[str, Dict[str, int]] = dict()
         for key in keys:
             metadata[key] = defaultdict(int)
 
         for osd in osd_map['osds']:
-            res = self.get_metadata('osd', str(osd['osd'])).items()
+            res = self.get_metadata('osd', str(osd['osd']))
             if res is None:
                 self.log.debug('Could not get metadata for osd.%s' % str(osd['osd']))
                 continue
-            for k, v in res:
+            for k, v in res.items():
                 if k not in keys:
                     continue
 
                 metadata[k][v] += 1
 
         return metadata
 
-    def gather_mon_metadata(self, mon_map):
+    def gather_mon_metadata(self,
+                            mon_map: Dict[str, List[Dict[str, str]]]) -> Dict[str, Dict[str, int]]:
         keys = list()
         keys += self.metadata_keys
 
-        metadata = dict()
+        metadata: Dict[str, Dict[str, int]] = dict()
         for key in keys:
             metadata[key] = defaultdict(int)
 
         for mon in mon_map['mons']:
-            res = self.get_metadata('mon', mon['name']).items()
+            res = self.get_metadata('mon', mon['name'])
             if res is None:
                 self.log.debug('Could not get metadata for mon.%s' % (mon['name']))
                 continue
-            for k, v in res:
+            for k, v in res.items():
                 if k not in keys:
                     continue
 
                 metadata[k][v] += 1
 
         return metadata
 
-    def gather_crush_info(self):
+    def gather_crush_info(self) -> Dict[str, Union[int,
+                                                   bool,
+                                                   List[int],
+                                                   Dict[str, int],
+                                                   Dict[int, int]]]:
         osdmap = self.get_osdmap()
         crush_raw = osdmap.get_crush()
         crush = crush_raw.dump()
 
-        def inc(d, k):
+        T = TypeVar('T', int, str)
```

Comment:
I'd be :+1: if we could the generic name  `T` only for generic type vars. Maybe something like `IntStrT`?

Reply:
renamed to `BucketKeyT`.